src/lexer.coffee

```diff
@@ -301,17 +301,59 @@ exports.Lexer = class Lexer
 
     end
 
-  # Matches and consumes comments.
+  # Matches and consumes comments. The comments are taken out of the token
+  # stream and saved for later, to be reinserted into the output after
+  # everything has been parsed and the JavaScript code generated.
   commentToken: ->
     return 0 unless match = @chunk.match COMMENT
     [comment, here] = match
+    contents = null
+    # Does this comment follow code on the same line?
+    newLine = /^\s*\n+\s*#/.test comment
     if here
-      if match = HERECOMMENT_ILLEGAL.exec comment
-        @error "block comments cannot contain #{match[0]}",
-          offset: match.index, length: match[0].length
-      if here.indexOf('\n') >= 0
-        here = here.replace /// \n #{repeat ' ', @indent} ///g, '\n'
-      @token 'HERECOMMENT', here, 0, comment.length
+      matchIllegal = HERECOMMENT_ILLEGAL.exec comment
+      if matchIllegal
+        @error "block comments cannot contain #{matchIllegal[0]}",
+          offset: matchIllegal.index, length: matchIllegal[0].length
+
+      # Parse indentation or outdentation as if this block comment didn’t exist.
+      chunk = @chunk.replace "####{here}###", ''
+      # Remove leading newlines, like `Rewriter::removeLeadingNewlines`, to
+      # avoid the creation of unwanted `TERMINATOR` tokens.
+      chunk = @chunk.replace /^\n+/, ''
+      @lineToken chunk
+
+      # Pull out the ###-style comment’s content, and format it.
+      content = here
+      content = content.replace /^(\s*)###/, ''
```

Comment:
if `here` contains the `###` prefix/suffix, doesn't it mean `chunk = @chunk.replace "####{here}###", ''` L320 is effectively doing nothing? (since it'd try to replace `######...######`